### PR TITLE
Python3 doctest compatibility: RemoveBackground to ResizeRectangularDetector

### LINKED ISSUE #20818: Python3 doctest compatibility: RemoveBackground to ResizeRectangularDetector
Python3 doctest compatibility: algorithms <RemoveBackground to ResizeRectangularDetector>

This issue is part of the move to Python3 in the doctests, 
Algorithms covered in this issue are:
```
RemoveBackground-v1
RemoveBins-v1
RemoveExpDecay-v1
RemoveLogs-v1
RemoveMaskedSpectra-v1
RemovePromptPulse-v1
RemoveWorkspaceHistory-v1
RenameLog-v1
RenameWorkspace-v1
RenameWorkspaces-v1
ReplaceSpecialValues-v1
ReplicateMD-v1
ResNorm-v1
ResNorm-v2
ResizeRectangularDetector-v1
```

### PR DESCRIPTION
**Pull request by brandonhewer**
Python3 doctest compatibility: algorithms <RemoveBackground  to ResizeRectangularDetector>

## Description of work.
This PR is part of refactoring effort to bring Python3 compatiblity into the doctests.
See https://github.com/mantidproject/mantid/issues/20672 for details.

The following tests were refactored to be compatible :
```
RemoveBackground-v1
RemoveBins-v1
RemoveExpDecay-v1
RemoveLogs-v1
RemoveMaskedSpectra-v1
RemovePromptPulse-v1
RemoveWorkspaceHistory-v1
RenameLog-v1
RenameWorkspace-v1
RenameWorkspaces-v1
ReplaceSpecialValues-v1
ReplicateMD-v1
ResNorm-v1
ResNorm-v2
ResizeRectangularDetector-v1
```

Code review.

Fixes #20818.
Internal change, no release notes

### CODE REVIEW

File: docs/source/algorithms/RemoveBackground-v1.rst [hunk 1]
```diff
@@ -206,9 +206,9 @@ Usage
    ER = Result .dataE(0);
    
    # print first spectra, Note invalid error calculations
-   print "| x sampl  | x result | S sample | S no bg  | Err samp | Err no_bg|"
+   print("| x sampl  | x result | S sample | S no bg  | Err samp | Err no_bg|")
    for i in xrange(0,20):
```
Comment (rosswhitfield) [inline docs/source/algorithms/RemoveBackground-v1.rst]: Should be `range` not `xrange`

File: docs/source/algorithms/RenameWorkspace-v1.rst [hunk 2]
```diff
@@ -42,34 +42,34 @@ Usage
...
    obj_inADS.sort()
-   print "{0:20}|{1:>6}| With Names: |{2:>20}|{3:>20}|".format("Exist in ADS: ",len(obj_inADS),obj_inADS[0],obj_inADS[1])
+   print("{0:20}|{1:>6}| With Names: |{2:>20}|{3:>20}|".format("Exist in ADS: ",len(obj_inADS),obj_inADS[0],obj_inADS[1]))
    # 
    mon_ws1 = NameB.getMonitorWorkspace()
-   print "The name of the monitor workspace attached to workspace:{0:>6}| Is:  {1:>10}|".\
-            format(NameB.name(),mon_ws1.name())
+   print("The name of the monitor workspace attached to workspace:{0:>6}| Is:  {1:>10}|".\
+            format(NameB.name(),mon_ws1.name()))
    print "*********************************************************************"
```
Comment (rosswhitfield) [inline docs/source/algorithms/RenameWorkspace-v1.rst]: You missed some `print` statements

File: docs/source/algorithms/ReplaceSpecialValues-v1.rst [hunk 3]
```diff
@@ -75,7 +75,7 @@ Output:
     ws = ReplaceSpecialValues(ws, SmallNumberThreshold=1e-6)
     
     print("Before\t\t After")
-    print("{0}\t{1}".format(wsYArray[0], ws.readY(0)[0]))
+    print("{0}\t{1:.1f}".format(wsYArray[0], ws.readY(0)[0]))
```
Comment (rosswhitfield) [inline docs/source/algorithms/ReplaceSpecialValues-v1.rst]: I get different precision here for python 3
```
Expected:
    Before                After
    9.99999993923e-09   0.0
Got:
    Before		 After
    9.99999993922529e-09	0.0
```

**Review by rosswhitfield — CHANGES_REQUESTED** [review]

**Comment by josephframsay** [comment]
Brandon asked me to take over these PRs as he's not here until 16th

**Review by rosswhitfield — APPROVED** [review]

**Pull request merged**